### PR TITLE
Added unique rack filter custom extension for quality of life improve…

### PR DESCRIPTION
**Pull request by djhoward12**
Added unique rack filter custom extension for quality of life improvement.

### CODE REVIEW

File: nautobot_floor_plan/forms.py [hunk 1]
```diff
@@ -136,9 +136,9 @@ class FloorPlanTileForm(NautobotModelForm):
 
     floor_plan = DynamicModelChoiceField(queryset=models.FloorPlan.objects.all())
     rack = DynamicModelChoiceField(
-        queryset=Rack.objects.all(),
-        required=False,
-        query_params={"nautobot_floor_plan_floor_plan": "$floor_plan", "rack_group": "$rack_group"},
+    queryset=Rack.objects.all(),
+    required=False,
+    query_params={"nautobot_floor_plan_floor_plan": "$floor_plan", "nautobot_floor_plan_unique_rack": False, "rack_group": "$rack_group"},
```
Comment (joewesch) [inline nautobot_floor_plan/forms.py:141]: Hmm, I'm curious why `black` didn't flag this as needing the leading whitespaces that were removed.

File: nautobot_floor_plan/filter_extensions.py [hunk 2]
```diff
@@ -19,6 +20,17 @@ class RackFilterExtension(PluginFilterExtension):
         ),
     }
 
+class UniqueRackFilterExtension(PluginFilterExtension):
+    """Add a filter to ensure a Rack is only an option if it isn't installed."""
+
+    model = "dcim.rack"
+
+    filterset_fields = {
+        "nautobot_floor_plan_unique_rack": RelatedMembershipBooleanFilter(
```
Comment (joewesch) [inline nautobot_floor_plan/filter_extensions.py:29]: Let's use the [core Best Practices](https://docs.nautobot.com/projects/core/en/stable/development/core/best-practices/#filter-naming-and-definition) here and use `has_floor_plan_tile` instead.

File: changes/117.added [hunk 3]
```diff
@@ -0,0 +1 @@
+Added unique rack filter_extension
```
Comment (joewesch) [inline changes/117.added:1]:
```suggestion
Added filter extension to be able to filter racks that are assigned/unassigned to a floor plan tile.
```

File: nautobot_floor_plan/filter_extensions.py [hunk 4]
```diff
@@ -19,6 +20,17 @@ class RackFilterExtension(PluginFilterExtension):
...
+    """Add a filter to ensure a Rack is only an option if it isn't installed."""
+
+    model = "dcim.rack"
+
+    filterset_fields = {
+        "nautobot_floor_plan_has_floor_plan_tile": RelatedMembershipBooleanFilter(
+            field_name="floor_plan_tile",
+            label="Floor plan",
+        ),
+    }
```
Comment (joewesch) [inline nautobot_floor_plan/filter_extensions.py:33]: Let's go ahead and add in the `filterform_fields` to add the filter to the UI as well.

Comment (djhoward12) [reply nautobot_floor_plan/filter_extensions.py:25]: verified the filter is present in Rack filters.

**Review by joewesch — APPROVED** [review]

**Pull request merged**